internal/metrics/metrics.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics holds all prometheus metrics
+type Metrics struct {
+	TotalLatency      *prometheus.HistogramVec
+	TokensTotal       *prometheus.CounterVec
+	RequestsTotal     *prometheus.CounterVec
+	FirstTokenLatency *prometheus.HistogramVec
+	InterTokenLatency *prometheus.HistogramVec
```

Comment:
Any exported fields must have an extensive comment, but this one especially needs them. See the other places in the code base and add comments like about exactly when this will be updated and how and which tags etc. I mean a really extensive comments without any ambiguity

Reply:
on the way

Reply:
comments added. Also added a markdown page to explain the metrics and usage.